AnalyzingAndProfilingTools/ProfileSample/ProfileSample/Controllers/HomeController.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Mvc;
+using ProfileSample.DAL;
+using ProfileSample.Models;
+
+namespace ProfileSample.Controllers
+{
+    public class HomeController : Controller
+    {
+        public async Task<ActionResult> Index()
+        {
+
+            //var sources = context.ImgSources.Take(20).Select(x => x.Id);
+
+            //foreach (var id in sources)
+            //{
+            //    var item = context.ImgSources.Find(id);
+
+            //    var obj = new ImageModel()
+            //    {
+            //        Name = item.Name,
+            //        Data = item.Data
+            //    };
+
+            //    model.Add(obj);
+            //} 
+
+            var temp = await Convert(); // First for added images db
+
+            var model = new List<ImageModel>();
+            var context = new ProfileSampleEntities();
+
+            model = context.ImgSources.Take(24).AsNoTracking()
+            .Select(x => new ImageModel()
+                {
+                    Name = x.Name,
+                    Data = x.Data
+                }).ToList();
+
+
+
+            return View(model);
+        }
+
+        public async Task<ActionResult> Convert()
+        {
+            var files = Directory.GetFiles(Server.MapPath("~/Content/Img"), "*.jpg");
+
+            using (var context = new ProfileSampleEntities())
+            {
+                foreach (var file in files)
+                {
+                    using (var stream = new FileStream(file, FileMode.Open))
+                    {
+                        byte[] buff = new byte[stream.Length];
+
+                        stream.Read(buff, 0, (int)stream.Length);

```

Comment:
either use async versions of Read `stream.ReadAsync` and context SaveChanges `context.SaveChangesAsync` methods, or make method synchronous https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/async-scenarios#important-info-and-advice

Reply:
Please check update comments commit